--- a/dlls/ntdll/unix/virtual.c
+++ b/dlls/ntdll/unix/virtual.c
@@ -7061,26 +7061,8 @@
 #ifdef __APPLE__
 
 /**********************************************************************
  *           NtFlushProcessWriteBuffers  (NTDLL.@)
  */
 NTSTATUS WINAPI NtFlushProcessWriteBuffers(void)
 {
-    static NTSTATUS (*p_thread_get_register_pointer_values)(thread_t, uintptr_t *, size_t *, uintptr_t *);
-    static pthread_once_t once = PTHREAD_ONCE_INIT;
-    mach_msg_type_number_t count, i;
-    thread_act_array_t threads;
-
-    pthread_once( &once, init_thread_get_register_pointer_values );
-    if (p_thread_get_register_pointer_values &&
-        task_threads( mach_task_self(), &threads, &count ) == KERN_SUCCESS)
-    {
-        uintptr_t reg_values[128];
-        size_t reg_count = ARRAY_SIZE( reg_values );
-        uintptr_t sp;
-
-        for (i = 0; i < count; i++)
-        {
-            if (threads[i] != mach_thread_self())
-                p_thread_get_register_pointer_values( threads[i], &sp, &reg_count, reg_values );
-            mach_port_deallocate( mach_task_self(), threads[i] );
-        }
-        vm_deallocate( mach_task_self(), (vm_address_t)threads, count * sizeof(threads[0]) );
-    }
+    __sync_synchronize();
     return STATUS_SUCCESS;
 }
