--- a/dlls/ntdll/signal_x86_64.c
+++ b/dlls/ntdll/signal_x86_64.c
@@ -877,6 +877,7 @@
     UNWIND_HISTORY_TABLE table;
     RUNTIME_FUNCTION *func;
     PEXCEPTION_ROUTINE handler;
+    NTSTATUS status;
     ULONG_PTR frame, base;
     CONTEXT context;
     void *data;
@@ -887,9 +888,18 @@
     for (i = 0; i < count; i++)
     {
         func = RtlLookupFunctionEntry( context.Rip, &base, &table );
-        if (RtlVirtualUnwind2( UNW_FLAG_NHANDLER, base, context.Rip, func, &context, NULL,
-                               &data, &frame, NULL, NULL, NULL, &handler, 0 ))
-            break;
+        status = STATUS_SUCCESS;
+        __TRY
+        {
+            status = RtlVirtualUnwind2( UNW_FLAG_NHANDLER, base, context.Rip, func, &context, NULL,
+                                        &data, &frame, NULL, NULL, NULL, &handler, 0 );
+        }
+        __EXCEPT_PAGE_FAULT
+        {
+            status = STATUS_ACCESS_VIOLATION;
+        }
+        __ENDTRY
+        if (status) break;
         if (!context.Rip) break;
         if (!frame || !is_valid_frame( frame )) break;
         if (context.Rsp == (ULONG_PTR)NtCurrentTeb()->Tib.StackBase) break;
